wagtail/snippets/views/snippets.py

```diff
@@ -68,24 +68,30 @@ def get_snippet_edit_handler(model):
 # == Views ==
 
 
-def index(request):
-    snippet_model_opts = [
-        model._meta
-        for model in get_snippet_models()
-        if user_can_edit_snippet_type(request.user, model)
-    ]
-    if snippet_model_opts:
-        return TemplateResponse(
-            request,
-            "wagtailsnippets/snippets/index.html",
-            {
-                "snippet_model_opts": sorted(
-                    snippet_model_opts, key=lambda x: x.verbose_name.lower()
-                )
-            },
-        )
-    else:
-        raise PermissionDenied
+class Index(IndexView):
```

Comment:
It looks like our `IndexView` has a lot of model-specific behaviour, so making this view inherit from that might make it tricker to change that view later.

Instead of extending from this ``IndexView``. Maybe we should directly extend from ``(WagtailAdminTemplateMixin, BaseListView)`` instead? (https://github.com/wagtail/wagtail/blob/main/wagtail/admin/views/generic/models.py#L56)

If there's anything that isn't model-specific that we need here, I think we should either repeat it or turn those into more mixins

Reply:
Well, even `BaseListView` extends from `MultipleObjectsMixin`, which also has quite a lot of model-specific behaviour (mainly pagination stuff). We also don't really use anything from `WagtailAdminTemplateMixin` here...

Should we just extend from `View` directly? (or maybe `WagtailAdminTemplateMixin` too, just in case)

Reply:
Yep, sounds like we should extend from ``View`` directly. Thanks for raising that!